Tests/SwiftSyntaxMacrosTest/MacroSystemTests.swift

```diff
@@ -726,6 +727,80 @@ public struct DeclsFromStringsMacroNoAttrs: DeclarationMacro {
   }
 }
 
+struct TracedPreambleMacro: PreambleMacro {
+  static func expansion(
+    of node: AttributeSyntax,
+    providingPreambleFor declaration: some DeclSyntaxProtocol & HasTrailingOptionalCodeBlock,
+    in context: some MacroExpansionContext
+  ) throws -> [CodeBlockItemSyntax] {
+    // FIXME: Should be able to support (de-)initializers and accessors as
+    // well, but this is a lazy implementation.
+    guard let funcDecl = declaration.as(FunctionDeclSyntax.self) else {
+      return []
+    }
+
+    let funcBaseName = funcDecl.identifier.text
+    let paramNames = funcDecl.signature.input.parameterList.map { param in
+      if let name = param.parameterName {
+        return name.text
+      } else {
+        return "_"
+      }
+    }
+
+    let passedArgs = paramNames.map { "\($0): \\(\($0))" }.joined()
+
+    let entry: CodeBlockItemSyntax = """
+      log("Entering \(raw: funcBaseName)(\(raw: passedArgs))")
+      """
+
+    let argLabels = paramNames.map { "\($0):" }.joined()
+
+    let exit: CodeBlockItemSyntax = """
+      log("Exiting \(raw: funcBaseName)(\(raw: argLabels))")
+      """
+
+    return [
+      entry,
+      """
+      defer {
+        \(exit)
+      }
+      """,
+    ]
+  }
+}
+
+struct RemoteBodyMacro: BodyMacro {
+  static func expansion(
+    of node: AttributeSyntax,
+    providingBodyFor declaration: some DeclSyntaxProtocol & HasTrailingOptionalCodeBlock,
+    in context: some MacroExpansionContext
+  ) throws -> [CodeBlockItemSyntax] {
+    // FIXME: Should be able to support (de-)initializers and accessors as
+    // well, but this is a lazy implementation.
+    guard let funcDecl = declaration.as(FunctionDeclSyntax.self) else {
+      return []
+    }
+
+    let funcBaseName = funcDecl.identifier.text
+    let paramNames = funcDecl.signature.input.parameterList.map { param in
+      if let name = param.parameterName {
+        return name.text
+      } else {
+        return "_"
+      }
+    }
+
+    let passedArgs = paramNames.map { "\"\($0)\": \($0)" }.joined(separator: ", ")
```

Comment:
What do you think about constructing the dictionary using SwiftSyntaxBuilder? It has a little less escaping that way and is easier to read IMO. To do this, `paramNames` should be an array of `TokenSyntax` instead of array of strings.

```swift
let d = DictionaryExprSyntax(content: .elements(DictionaryElementListSyntax {
  for paramName in paramNames {
    DictionaryElementSyntax(
      keyExpression: ExprSyntax("\(literal: paramName.text)"),
      valueExpression: IdentifierExprSyntax(identifier: paramName)
    )
  }
}))
```

Reply:
I tend to like my string interpolations, but... sure. I can do this.